cacheDependencyManagers/composerConfig.js

```diff
@@ -5,8 +5,22 @@ var path = require('path');
 var shell = require('shelljs');
 var logger = require('../util/logger');
 var md5 = require('md5');
-var composerFilePath = path.resolve(process.cwd(), 'composer.json');
+var isUsingComposerLock = false;
 
+// Returns path to configuration file for composer. Uses
+// composer.lock if it exists; otherwise,
+// defaults to composer.json
+var getComposerConfigPath = function () {
+  var composerLockPath = path.resolve(process.cwd(), 'composer.lock');
+  var composerJsonPath = path.resolve(process.cwd(), 'composer.json');
+
+  if (fs.existsSync(composerLockPath)) {
+    logger.logInfo('[composer] using composer.lock instead of composer.json');
+    isUsingComposerLock = true;
+  }
+
+  return isUsingComposerLock ? composerLockPath : composerJsonPath;
+};
```

Comment:
Like so:

```
var isUsingComposerLock = true;

// Returns path to configuration file for composer. Uses
// composer.lock if it exists; otherwise,
// defaults to composer.json
var getComposerConfigPath = function () {
  var composerLockPath = path.resolve(process.cwd(), 'composer.lock');
  var composerJsonPath = path.resolve(process.cwd(), 'composer.json');

  if (! fs.existsSync(composerLockPath)) {
    logger.logInfo('[composer] using composer.json instead of composer.lock');
    isUsingComposerLock = false;
  }

  return ! isUsingComposerLock ? composerJsonPath : composerLockPath;
};
```